specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Workspaces.json

```diff
@@ -272,6 +272,75 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.OperationalInsights/deletedWorkspaces": {
+      "get": {
+        "tags": [
+          "DeletedWorkspaces"
+        ],
+        "x-ms-examples": {
+          "WorkspacesSubscriptionList": {
+            "$ref": "./examples/WorkspacesSubscriptionList.json"
+          }
+        },
+        "operationId": "DeletedWorkspaces_List",
+        "description": "Gets recently deleted workspaces in a subscription, available for recovery.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK response definition.",
+            "schema": {
+              "$ref": "#/definitions/WorkspaceListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/deletedWorkspaces": {
+      "get": {
+        "tags": [
+          "DeletedWorkspaces"
+        ],
+        "x-ms-examples": {
+          "WorkspacesGet": {
+            "$ref": "./examples/WorkspacesListByResourceGroup.json"
+          }
+        },
+        "operationId": "DeletedWorkspaces_ListByResourceGroup",
+        "description": "Gets recently deleted workspaces in a resource group, available for recovery.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK response definition.",
+            "schema": {
+              "$ref": "#/definitions/WorkspaceListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
```

Comment:
>null [](start = 26, length = 4)

You should consider declaring paging in the swagger even if the service doesn't support it. This way you can just add it in the service when needed and clients will just work the same. (Same for the above API as well.)

Reply:
We consider adding such support in the future for Workspaces, we will add it then for DeletedWorkspaces as well, since it is the same code in the service. We prefer to keep it this way for now.

Reply:
I'm only suggesting a swagger change. If your backend doesn't support paging it's as if it always returns 1 page, but it's not required at this point.

---
In reply to: [422477036](https://github.com/Azure/azure-rest-api-specs/pull/9401#discussion_r422477036) [](ancestors = 422477036)